### ISSUE TITLE
Can't parse Stmt::Expr

### ISSUE THREAD

**Issue by okamt** [dtolnay/syn#1731]
Not sure if I'm doing something wrong, but I can't get syn to parse a Stmt::Expr without a semicolon. It panics with `unexpected end of input, expected semicolon`.

Rust playground link: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=cbc75065fde246a57eda32f48d40bddd

**Comment by a101010** [comment]
Yes, this should work. Final expression without semicolon is important. Is there a workaround or another way to parse this?

use syn; // 2.0.79

fn main() {
    let return_stmt = syn::parse_str::<syn::Stmt>("2")
        .expect("failed to parse return statment");
    println!("{return_stmt:?}");
}

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a89e0abe4ec8a0314dda0e1b077f7011

**Comment by a101010** [comment]
It should get parsed as sym::Stmt::Expr(Expr, None)

**Comment by a101010** [comment]
Oh, jaja, I guess the workaround is pretty simple. Just parse as expression and wrap it explicitly.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ad09ef9cba00784aa8506eb2e0feb4e7
